### ISSUE TITLE
Received a QuantityType '1.12 W' with unit, but the boundaries are defined as a plain number without units

### ISSUE THREAD

**Issue by spacemanspiff2007** [openhab/openhab-core#3344]
I'm still on OH3.3 but I read through the change log and found nothing concerning the range profile. If this is already fixed just close the issue.

I've defined the following item:
```
Switch   RangeSwitch    "Active [%s]"   { channel="<CHANNEL>" [profile="system:range", lower="0 W", upper="9.2 W", inverted="true"]}
```

Every time a new value is received the following line gets logged:
```
[nal.profiles.SystemRangeStateProfile] - Received a QuantityType '1.07 W' with unit, but the boundaries are defined as a plain number without units (lower=0, upper=9.2), please consider adding units to them.
```

RestAPI returns proper config:
```
    "configuration": {
      "inverted": "true",
      "profile": "system:range",
      "lower": "0 W",
      "upper": "9.2 W"
    },
```

**Comment by J-N-K** [comment]
Is this still the case with latest code? I believe it is the same that was fixed with #3526.

**Comment by J-N-K** [comment]
@spacemanspiff2007 Can you check if this is still the case?

**Comment by spacemanspiff2007** [comment]
I'm not on OH4 yet and I am hesitant to already migrate my production system.

I tried to reproduce the issue with an mqtt item and a local broker.
First I created the item with the profile without units.
I can both publish `20` and `20 W` and the switch turns on without issues.

Then I modified the profile so it uses UoM values as boundaries.
Publishing `20 W` works as expected, for `20` however I get the following message:
````
Incoming payload '20' not supported by type 'NumberValue'
````

Once I added `unit="W",` to the item metadata that works also.
The original issue is fixed, but the error message definitely needs to be improved.
I know what the issue was because I was involved with the UoM discussion but for someone else this message will make no sense.
